### PR TITLE
Add BatchRequestCore

### PR DESCRIPTION
**Pull request by david-eckardt-sociomantic**
The `BatchRequestCore` module contains the controller for all-nodes requests where records are sent in batches so suspending happens only in the client.

### CODE REVIEW

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: Param doesn't exist.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: These "Tells the nodes" comments are misleading now. Maybe change to something like "Stops the flow of data to this request."?

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d:185]: Maybe also state that further data may be passed to the user, even after stop has been called.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: The gods of line-wrapping do not look kindly upon this sort of behaviour.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: `batchRequestConnector` and `BatchRequestInitialiser` are mentioned here but do not appear in this module. I guess this comment is cut & pasted from the old suspendable core stuff and these things are no longer needed / relevant.

**Review by gavin-norman-sociomantic — COMMENTED** [review]
A few doc / style comments, but otherwise looks good!

**Comment by gavin-norman-sociomantic** [comment]
Marked as blocked until the whole chain of PRs (this, dmqproto, dmqnode) is confirmed to be basically working.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (david-eckardt-sociomantic) [reply src/swarm/neo/client/mixins/BatchRequestCore.d]: Yes, I removed most mentions from the comments but didn't catch all of them.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (david-eckardt-sociomantic) [reply src/swarm/neo/client/mixins/BatchRequestCore.d]: :scream_cat:

**Comment by david-eckardt-sociomantic** [comment]
Updated, addressing documentation/style comments.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: "by and"

**Comment by david-eckardt-sociomantic** [comment]
Updated.

**Review by gavin-norman-sociomantic — APPROVED** [review]
LGTM, but leaving blocked until we want to merge the whole chain.

**Comment by gavin-norman-sociomantic** [comment]
I think this will need to be updated with the additional connector logic that you added yesterday, right @david-eckardt-sociomantic ?

**Comment by david-eckardt-sociomantic** [comment]
Updated, adding `batchRequestConnector`.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: I _think_ this should say "To be called from the Connector policy instance of an AllNodesRequest."

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: Argument doesn't exist.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: Should be `batchRequestConnector`. (How does this compile?)

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/mixins/BatchRequestCore.d]: (Ah, it doesn't ;)

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d:8]: `batchRequestConnector` should be mentioned in this list too.

**Comment by david-eckardt-sociomantic** [comment]
Updated, addressing the comments for `batchRequestConnector`.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/BatchRequestCore.d]: "Forwards to batchRequestConnector"

Comment (david-eckardt-sociomantic) [reply src/swarm/neo/client/mixins/BatchRequestCore.d]: Thanks.

**Comment by gavin-norman-sociomantic** [comment]
As the first round of testing with Consume v3 has been done, unblocking and merging. We can make patch releases, if we discover more issues.

File: src/swarm/neo/client/mixins/BatchRequestCore.d
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/mixins/BatchRequestCore.d]: This still.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/mixins/BatchRequestCore.d:8]: This still.

**Review by gavin-norman-sociomantic — CHANGES_REQUESTED** [review]
One last point. Could you also squash the two commits?

**Comment by david-eckardt-sociomantic** [comment]
Updated, mentioning `batchRequestConnector` in the module header documentation and squashing the two commits.

**Review by gavin-norman-sociomantic — APPROVED** [review]

**Pull request merged**
